core/linux.go

```diff
@@ -316,11 +316,13 @@ func (l *library) CompileObjs(ctx blueprint.ModuleContext) []string {
 		buildWrapper, buildWrapperDeps := l.Properties.Build.getBuildWrapperAndDeps(ctx)
 		args["build_wrapper"] = buildWrapper
 
-		sourceWithoutBuildDir := source
-		if buildDir := g.buildDir(); strings.HasPrefix(sourceWithoutBuildDir, buildDir) {
-			sourceWithoutBuildDir = sourceWithoutBuildDir[len(buildDir):]
+		output := l.ObjDir() + source + ".o"
+		if buildDir := g.buildDir(); strings.HasPrefix(source, buildDir) {
```

Comment:
Pref: This is a bit hard to follow - you've got an `else` clause, but are also overwriting stuff in the `if` part.

Maybe it could be written as:
```go
var sourceWithoutPrefix string
if buildDir := g.buildDir(); strings.HasPrefix(source, buildDir) {
    sourceWithoutPrefix = source[len(buildDir):]
    // `source` already has a buildDir prefix
} else {
    sourceWithoutPrefix = source
    source = filepath.Join(g.sourcePrefix(), source)
}
var output = l.ObjDir() + sourceWithoutPrefix + ".o"
```
(approximately)

Reply:
Ok sure if this is hard to follow then I've changed it to be more readable